cmd/snap/cmd_debug_state.go

```diff
@@ -227,6 +234,201 @@ func (c *cmdDebugState) showIsSeeded(st *state.State) error {
 	return nil
 }
 
+// connState is the connection state stored by InterfaceManager
+type connState struct {
+	Auto             bool                   `json:"auto,omitempty" yaml:"auto"`
+	ByGadget         bool                   `json:"by-gadget,omitempty" yaml:"by-gadget"`
+	Interface        string                 `json:"interface,omitempty" yaml:"interface"`
+	Undesired        bool                   `json:"undesired,omitempty" yaml:"undesired"`
+	StaticPlugAttrs  map[string]interface{} `json:"plug-static,omitempty" yaml:"plug-static,omitempty"`
+	DynamicPlugAttrs map[string]interface{} `json:"plug-dynamic,omitempty" yaml:"plug-dynamic,omitempty"`
+	StaticSlotAttrs  map[string]interface{} `json:"slot-static,omitempty" yaml:"slot-static,omitempty"`
+	DynamicSlotAttrs map[string]interface{} `json:"slot-dynamic,omitempty" yaml:"slot-dynamic,omitempty"`
+	// TODO: hotplug
+}
+
+type connectionInfo struct {
+	PlugSnap string
+	PlugName string
+	SlotSnap string
+	SlotName string
+
+	connState
+}
+
+type byPlug []*connectionInfo
+
+func (c byPlug) Len() int      { return len(c) }
+func (c byPlug) Swap(i, j int) { c[i], c[j] = c[j], c[i] }
+func (c byPlug) Less(i, j int) bool {
+	a := c[i]
+	b := c[j]
+
+	if a.PlugSnap < b.PlugSnap {
+		return true
+	}
+
+	if a.PlugSnap == b.PlugSnap {
+		if a.PlugName < b.SlotName {
+			return true
+		}
+	}
+	return false
+}
+
+func parsePlugOrSlot(plugOrSlot string) (snap, name string, err error) {
+	p := strings.Split(plugOrSlot, ":")
+	switch len(p) {
+	case 2:
+		snap, name = p[0], p[1]
+	case 1:
+		snap = p[0]
+	default:
+		return "", "", fmt.Errorf("cannot parse %q", plugOrSlot)
+	}
+	return snap, name, nil
+}
+
+func parseConnID(connID string) (plugSnap, plugName, slotSnap, slotName string, err error) {
```

Comment:
I would suggest moving this parser to a package that can be shared by both snapd and snap. Otherwise we should reference the original code.

Reply:
Good point, I actually reused ConnRef parsing from interfaces, I thought pulling interfaces core should be avoided but we already do that in cmd/snap.